docs/configuration-files.md

```diff
@@ -272,91 +276,90 @@ rules:
 
 This rule looks for usage of Django's [`FloatField`](https://docs.djangoproject.com/en/3.0/ref/models/fields/#django.db.models.FloatField) model when storing currency information. `FloatField` can lead to rounding errors and should be avoided in favor of [`DecimalField`](https://docs.djangoproject.com/en/3.0/ref/models/fields/#django.db.models.DecimalField) when dealing with currency. Here the `pattern-where-python` operator allows us to utilize the Python `in` statement to filter findings that look like currency.
 
-## Metavariable matching
+## Metavariable Matching
 
-### Metavariables in logical inclusions
+Metavariable matching operates differently for logical AND (`patterns`)
+and logical OR (`pattern-either`) parent operators. Note that the behavior is
+consistent across all child operators: `pattern`, `pattern-not`,
+`pattern-regex`, `pattern-inside`, `pattern-not-inside`.
 
-Patterns' matched metavariable values are enforced to be identical when performing logical inclusion operations (`patterns`, `pattern-inside`) on matches.
+### Metavariables in Logical ANDs
+
+Metavariable values must be identical across sub-patterns when performing
+logical AND operations with the `patterns` operator.
 
 **Example**
 
-Consider the configuration:
+Consider the following rule:
+
 ```yaml
-  patterns:
-    - pattern-inside: |
-        def $F($X):
-            ...
-    - pattern: open($X)
+rules:
+  - id: function-args-to-open
+    patterns:
+      - pattern-inside: |
+          def $F($X):
+              ...
+      - pattern: open($X)
+    message: "Function argument passed to open() builtin"
+    languages: [python]
+    severity: ERROR
 ```
 
-This configuration will match this Python code:
+This rule will match the following code:
+
 ```python
 def foo(path):
     open(path)
 ```
 
 But will not match this code:
+
 ```python
 def foo(path):
     open(something_else)
 ```
 
-### Metavariables in logical negations
```

Comment:
I lumped negations in with "Note that the behavior is consistent across all child operators: `pattern`, `pattern-not`, `pattern-regex`, `pattern-inside`, `pattern-not-inside`." above. Does that make sense?

Reply:
I think the example is useful, so I'd probably leave it in its own section 🤷

Reply:
Does it make sense to include it as another example under the `patterns` metavariable matching section?

Reply:
sure